avocado/core/test.py

```diff
@@ -1093,13 +1094,40 @@ def fetch_asset(self, name, asset_hash=None, algorithm=None,
         :param locations: list of URLs from where the asset can be
                           fetched (optional)
         :param expire: time for the asset to expire
-        :raise EnvironmentError: When it fails to fetch the asset
-        :returns: asset file local path
+        :param find_only: whether it should only look for the asset in the
+                          cache and do not download/move it to the cache.
+        :param cancel_on_missing: whether the test should be canceled if the
+                                  asset was not found in the cache or if
+                                  `fetch` could not add the asset to the cache.
+        :raises: :class:`exceptions.TestCancel when `cancel_on_missing` is
```

Comment:
The `TestCancel` and similar exceptions are really a way for the runner to be notified, and not really something that users should handle (and know about). So, while this is true, I think this should:
1) use `self.cancel()` instead (because the *core* implementation that raises exception is just that, an implementation), and `self.cancel()` is a public and supported API
2) Do not document this exception as it's not intended for test writers to catch it

Reply:
Okay, changed on v3!